src/coresyftools/coresyftools/tool_tester.py

```diff
@@ -0,0 +1,168 @@
+import sys
+import os
+import logging
+import subprocess
+from os.path import exists, getsize
+from argument_parser import CoReSyFArgumentParser
+from manifest import get_manifest
+
+
+class InvalidCommandException(Exception):
+    pass
+
+
+class ExampleTitleMissingException(Exception):
+    pass
+
+
+class ExampleDescriptionMissingException(Exception):
+    pass
+
+
+class ToolExampleCommand(object):
+
+    def __init__(self, manifest, title, description, command):
+        self.title = title
+        self.description = description
+        self.command = command
+        self.manifest = manifest
+        self._parse_arguments()
+
+    def _parse_arguments(self):
+        self.arg_parser = CoReSyFArgumentParser(self.manifest)
+        self.arg_parser.parse_arguments(self.command[1:])
+        self.outputs = [self.arg_parser.bindings[arg] for arg in
+                        self.arg_parser.outputs]
+
+    def run(self):
+        proc = subprocess.Popen(self.command, stdin=subprocess.PIPE,
+                                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = proc.communicate()
+        return (proc.returncode, stdout, stderr)
+
+    def __str__(self):
+        return ' '.join(self.command)
+
+
+def normalize_variable_name(name):
+    return name
+
+
+class ToolTester(object):
+
+    def __init__(self, tool_dir):
+        self.tool_dir = tool_dir
+        examples_file = 'examples.sh'
+        self.manifest_file_name = os.path.join(tool_dir, 'manifest.json')
+        self.manifest = get_manifest(self.manifest_file_name)
+        self._load_examples_file(examples_file)
+        self.logger = logging.getLogger()
+        self.logger.addHandler(logging.StreamHandler(sys.stdout))
+        self.logger.setLevel(logging.DEBUG)
+
+    def _load_examples_file(self, file_name):
+        cwd = os.getcwd()
+        os.chdir(self.tool_dir)
+        try:
+            with open(file_name) as examples_file:
+                self.example_commands = []
+                title = None
+                description = ''
+                ln = 0
+                for line in examples_file:
```

Comment:
I think it would be more confusing, since we will need to re-assign the local variables `title`, `description`, and `line_number` in the body of the function.